File: 005-TypeScript/000-Exercícios/Projeto L1E1 - OO/src/pessoa.ts
export class Pessoa{

    //Atributos
    private idade:number; 
    private diaNasc:number; 
    private mesNasc:number; 
    private anoNasc:number; 
    private nome: string; 

    //Construtor 
    public constructor(idade: number, diaNasc: number, 
        mesNasc:number, anoNasc: number, nome: string){
            this.idade = idade; 
            this.diaNasc = diaNasc; 
            this.mesNasc = mesNasc;
            this.anoNasc = anoNasc; 
            this.nome = nome; 
    }


    //Getters e Setters 
    public getIdade(): number{
        return this.idade; 
    }

    public getNome(): string{
        return this.nome; 
    }

    //Método de trabalho  
    public setDataDeNascimento(dia:number, mes:number, ano:number):void{
        this.diaNasc = dia;
        this.mesNasc = mes; 
        this.anoNasc = ano; 
    }

    public calculaIdade(dia:number, mes:number, ano:number):void{
        let diaAtual:Date = new Date(`${dia}/${mes}/${ano} 00:00:00`) 
        let diaNasc:Date = new Date(`${this.diaNasc}/${this.mesNasc}/${this.anoNasc} 00:00:00`) 
        
        this.idade = ((((((diaAtual.valueOf() - diaNasc.valueOf())/1000)/60)/60)/24)/365);
    }
}

File: 008-Java/001- Exemplos/006 - POO - Exemplo Pessoa/src/Pessoa.java

public class Pessoa {

	//Atributos 
	private int idade; 
	private byte dia; 
	private byte mes; 
	private short ano;
	private String nome; 
	
	//Construtor 
	Pessoa(String nome, int dia, int mes, int ano, int anoAtual){
		this.nome = nome; 
		this.dia = (byte)dia; 
		this.mes = (byte)mes; 
		this.ano = (short)ano; 
		calculaIdade((short)anoAtual); 
	}
	
	
	//Métodos de trabalho
	private void calculaIdade(short anoAtual) {
		this.idade =  anoAtual - this.ano; 
	}
	
	//Método para consultar a idade da pessoa 
	public int getIdade() {
		return this.idade; 
	}
	
	//Método para consultar o nome 
	public String getNome() {
		return this.nome; 
	}
	
	//Ajusta Data de Nascimento
	public void ajustaDataNascimento(int dia, int mes, int ano, int anoAtual) {
		this.dia = (byte)dia; 
		this.mes = (byte)mes; 
		this.ano = (short)ano; 
		calculaIdade((short)anoAtual); 
	}
	
	
	
}
